### DISCUSSION TITLE
Single target for multiple apps

### DISCUSSION THREAD

**Discussion by trev91** [fastlane/fastlane#20093] [Q&A]
My intention is to keep a very small xcode project and interact with xcode the least possible.

My hope is that I can create a generic "base" target to build apps with. Then, using the fastlane `download` action to retrieve a list of JSON objects from my server, I'll have a list of objects that contain values I'd inject into the Info.plist file for the generic "base" target. That same JSON object (per app) would include information like the developer team ID, bundle display name, and other app-specific information. I plan to use match and pem to help manage codesigning and certificates. The last piece is making sure that the app knows what Launch Screen and App icon to use - which I'm sure I could figure out.

Is it possible that this approach could result in us pushing out multiple apps from a single target, all from the same codebase? Could I really build apps like this at runtime of the fastlane lane?

**Comment by lucgrabowski** [comment]
It sounds possible but it will require writing scripts. 
Before running `match` and `gym` you could have a script that prepares the project: fetches data from your server, updates project, copies icons and assets. Think which of those things you could store in code repository.

And just to show how it could look like:
```ruby
<prepare_app>
match
gym
reset_git_repo 
...
<prepare_app>
match
gym
reset_git_repo
```
Of course it would be good to add helper lanes grouping similar calls. After one app is built you should reset git repository (if you don't create branches and commit changes per app) https://docs.fastlane.tools/actions/reset_git_repo/

You could also consider using dot env files
https://gist.github.com/joshdholtz/cd077ccdddb608af00eb2466259e7ac0
https://docs.fastlane.tools/best-practices/keys/
